custom_components/localtuya/cover.py

```diff
@@ -81,72 +99,68 @@ def __init__(
     @property
     def supported_features(self):
         """Flag supported features."""
-        supported_features = (
-            SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_SET_POSITION
-        )
+        supported_features = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP
+        if self._config[CONF_POSITIONING_MODE] != CONF_MODE_NONE:
+            supported_features = supported_features | SUPPORT_SET_POSITION
         return supported_features
 
     @property
     def current_cover_position(self):
         """Return current cover position in percent."""
-        # self.update()
-        # state = self._state
-        #        _LOGGER.info("curr_pos() : %i", self._position)
-        # print('curr_pos() : state [{}]'.format(state))
-        return self._position
+        return self._current_cover_position
 
     @property
     def is_opening(self):
         """Return if cover is opening."""
-        # self.update()
         state = self._state
-        # print('is_opening() : state [{}]'.format(state))
-        if state == "on":
-            return True
-        return False
+        return state == self._config[CONF_OPEN_CMD]
 
     @property
     def is_closing(self):
         """Return if cover is closing."""
-        # self.update()
         state = self._state
-        # print('is_closing() : state [{}]'.format(state))
-        if state == "off":
-            return True
-        return False
+        return state == self._config[CONF_CLOSE_CMD]
+
+    @property
+    def is_open(self):
+        """Return if the cover is open or not."""
+        if self._config[CONF_POSITIONING_MODE] != CONF_MODE_POSITION:
+            return None
+        else:
```

Comment:
You don't need this else here due to return. There's usually a lint warning about this...

Reply:
> Ok, so, the schema validation problem happens because `float` validates [0, 1[ for some reason. 

Gosh, I could have never imagined that. How did you figure this out, for curiosity??? The check_config script does not give validation errors...